lib/medtronic/medtronicSimulator.js

```diff
@@ -65,7 +83,220 @@ exports.make = function(config){
+          checkForScheduleChanges(event);
+          currBasal.with_expectedDuration(currBasal.duration);
+          var duration = Date.parse(event.time) - Date.parse(currBasal.time);
+          currBasal.with_duration(duration);
+
+          currBasal = currBasal.done();
+          events.push(currBasal);
+          setCurrBasal(null);
+          return;
+        }
+
+        if(currBasal.deliveryType === 'temp' && event.deliveryType === 'temp' && event.duration !== 0) {
+          // temp basal was updated
+          currBasal.with_expectedDuration(currBasal.duration)
+                   .with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          event.suppressed = _.clone(currBasal.suppressed);
+        }
+
+        if(event.deliveryType === 'temp') {
+
+          if(currBasal.deliveryType === 'scheduled') {
+            var suppressed = {
+              type: 'basal',
+              deliveryType: 'scheduled',
+              rate: currBasal.rate
+            };
+            event.suppressed = suppressed;
+
+            if (currPumpSettings) {
+              currBasal.with_scheduleName(currPumpSettings.activeSchedule);
+              event.suppressed.scheduleName = currBasal.scheduleName;
+            }
+          }
+        }
+
+        if(event.deliveryType === 'suspend') {
+          event.suppressed = {
+            type: 'basal',
+            deliveryType: currBasal.deliveryType,
+            rate: currBasal.rate
+          };
+          if (currPumpSettings && event.suppressed.deliveryType === 'scheduled') {
+            event.suppressed.scheduleName = currPumpSettings.activeSchedule;
+          }
+          if(currBasal.deliveryType === 'temp') {
+            // nest suppressed scheduled basal in temp basal
+            event.suppressed.suppressed = currBasal.suppressed;
+
+            // also, restart temp basal on resume
+            if(currStatus && currStatus.index === event.index) {
+              // if the indexes are the same, the suspended basal was
+              // created from the same record as the suspend/resume event
+              event.duration = currStatus.duration;
+
+              var resumeBasal = _.clone(currBasal);
+              resumeBasal.time = new Date(Date.parse(currStatus.time) + currStatus.duration).toISOString();
+              resumeBasal.deviceTime = sundial.formatDeviceTime(Date.parse(currStatus.deviceTime) + currStatus.duration);
+              resumeBasal.index = event.resumeIndex;
+              delete resumeBasal.duration; // we don't know the new duration yet
+
+              // finish up the temp basal before the suspend
+              if(currBasal.duration) {
+                currBasal.with_expectedDuration(currBasal.duration);
+                var duration = Date.parse(event.time) - Date.parse(currBasal.time);
+                currBasal.with_duration(duration);
+
+                resumeBasal.duration = currBasal.expectedDuration - currBasal.duration - currStatus.duration;
```

Comment:
I implemented "restart temp basal on resume" after seeing that the CareLink driver does resume temp basals after a manually entered `suspend`. Just double-checked on the pump UI itself and the temp basal does indeed resume after a suspend. Phew, you had me worried for a bit there ;)